src/odb/src/db/dbPowerSwitch.cpp

```diff
@@ -256,31 +254,35 @@ void dbPowerSwitch::addPortMap(const std::string& model_port,
                                const std::string& switch_port)
 {
   _dbPowerSwitch* obj = (_dbPowerSwitch*) this;
-  dbMaster* master = (dbMaster*) dbMaster::getMaster(
-      dbLib::getLib((dbDatabase*) obj->getImpl()->getDatabase(), obj->_lib),
-      obj->_lib_cell);
+  dbMaster* master = getLibCell();
 
+  auto logger = obj->getImpl()->getLogger();
   if (!master) {
-    obj->getImpl()->getLogger()->error(
-        utl::ODB,
-        32001,
-        "Cannot map port {} to {} because no lib cell is added",
-        model_port,
-        switch_port);
+    logger->error(utl::ODB,
+                  32001,
+                  "Cannot map port {} to {} because no lib cell is added",
+                  model_port,
+                  switch_port);
     return;
   }
 
   dbMTerm* mterm = master->findMTerm(switch_port.c_str());
+  if (mterm == nullptr) {
+    logger->error(utl::ODB,
+                  33002,
```

Comment:
Logger codes should be at most 4 digits.

Reply:
Is this a hard requirement, or just a suggestion?

Reply:
It is a strong suggestion unless you have a reason why you need five digits. We hope to keep uniformity across the tool; all other tools use four digits, and our code automatically pads them with `0` if needed, i.e., `1` --> `0001`. Also, `9999` should be plenty of messages 😉 .

Reply:
alright, I've changed as many as I could for this PR. If I come across any others that I've done I'll change them in another PR.